org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/popup/GradientColors.java

```diff
@@ -0,0 +1,193 @@
+    RGB bg = getBackground();
+    RGB bottom;
+    RGB top;
+
+    if (testTwoPrimaryColors(titleBg, 179, 256)) {
+      // Group 1
+      // Rule: If at least 2 of the RGB values are equal to or between 180 and
+      // 255, then apply specified opacity for Group 1
+      // Examples: Vista, XP Silver, Wn High Con #2
+      // Gradient Bottom = TITLE_BACKGROUND @ 30% Opacity over LIST_BACKGROUND
+      // Gradient Top = TITLE BACKGROUND @ 0% Opacity over LIST_BACKGROUND
+      bottom = blend(titleBg, bg, 30);
+      top = bg;
+    } else if (testTwoPrimaryColors(titleBg, 120, 180)) {
+      // Group 2
+      // Rule: If at least 2 of the RGB values are equal to or between 121 and
+      // 179, then apply specified opacity for Group 2
+      // Examples: XP Olive, OSX Graphite, Linux GTK, Wn High Con Black
+      // Gradient Bottom = TITLE_BACKGROUND @ 20% Opacity over LIST_BACKGROUND
+      // Gradient Top = TITLE BACKGROUND @ 0% Opacity over LIST_BACKGROUND
+      bottom = blend(titleBg, bg, 20);
+      top = bg;
+    } else {
+      // Group 3
+      // Rule: If at least 2 of the RGB values are equal to or between 0 and
+      // 120, then apply specified opacity for Group 3
+      // Examples: XP Default, Wn Classic Standard, Wn Marine, Wn Plum, OSX
+      // Aqua, Wn High Con White, Wn High Con #1
+      // Gradient Bottom = TITLE_BACKGROUND @ 10% Opacity over LIST_BACKGROUND
+      // Gradient Top = TITLE BACKGROUND @ 0% Opacity over LIST_BACKGROUND
+      bottom = blend(titleBg, bg, 10);
+      top = bg;
+    }
+
+    gradientBegin = getColor(resourceManager, top);
+    gradientEnd = getColor(resourceManager, bottom);
+  }
+
+  private static RGB blend(RGB c1, RGB c2, int ratio) {
+    int r = blend(c1.red, c2.red, ratio);
+    int g = blend(c1.green, c2.green, ratio);
+    int b = blend(c1.blue, c2.blue, ratio);
+    return new RGB(r, g, b);
+  }
+
+  private static int blend(int v1, int v2, int ratio) {
+    int b = (ratio * v1 + (100 - ratio) * v2) / 100;
+    return Math.min(255, b);
+  }
+
+  private boolean testTwoPrimaryColors(RGB rgb, int from, int to) {
+    int total = 0;
+    if (testPrimaryColor(rgb.red, from, to)) {
+      total++;
+    }
+    if (testPrimaryColor(rgb.green, from, to)) {
+      total++;
+    }
+    if (testPrimaryColor(rgb.blue, from, to)) {
+      total++;
+    }
+    return total >= 2;
+  }
+
+  private static boolean testPrimaryColor(int value, int from, int to) {
+    return value > from && value < to;
+  }
+
+  private RGB getSystemColor(int code) {
+    return getDisplay().getSystemColor(code).getRGB();
+  }
+
+  private Color getImpliedBackground() {
+    return display.getSystemColor(SWT.COLOR_LIST_BACKGROUND);
+  }
+
+  private Display getDisplay() {
+    return display;
+  }
+
+  private static Color getColor(ResourceManager manager, RGB rgb) {
+    try {
+      return manager.createColor(rgb);
+    } catch (DeviceResourceException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1166)